overlay/hermes-static-link.patch

```diff
@@ -0,0 +1,145 @@
+diff --git a/API/hermes/CMakeLists.txt b/API/hermes/CMakeLists.txt
+index 0b7cbacfe..df02e0d0b 100644
+--- a/API/hermes/CMakeLists.txt
++++ b/API/hermes/CMakeLists.txt
+@@ -59,7 +59,7 @@ add_hermes_library(traceInterpreter TraceInterpreter.cpp
+ 
+ set(HERMES_LINK_COMPONENTS LLVHSupport)
+ 
+-add_library(libhermes SHARED ${api_sources})
++add_library(libhermes ${api_sources})
+ 
+ # This is configured using a cmake flag instead of a separate target, because
+ # we need the output to be named "libhermes.so".
+@@ -76,6 +76,15 @@ target_link_libraries(libhermes
+ )
+ target_link_options(libhermes PRIVATE ${HERMES_EXTRA_LINKER_FLAGS})
+ 
++if (NOT BUILD_SHARED_LIBS)
++  bundle_static_library(hermesbundled libhermes jsi)
++
++  install(
++    FILES ${CMAKE_CURRENT_BINARY_DIR}/libhermesbundled.a
++    DESTINATION lib
++  )
++endif ()
++
+ # Export the required header directory
+ target_include_directories(libhermes PUBLIC .. ../../public ${HERMES_JSI_DIR})
+ 
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 9a7b8bd93..c9ba00336 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -290,6 +290,8 @@ set(HERMES_BUILD_LEAN_LIBHERMES OFF CACHE BOOL "Exclude the Hermes compiler from
+ 
+ set(HERMES_BUILD_SHARED_JSI OFF CACHE BOOL "Build JSI as a shared library.")
+ 
++option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
++
+ if (HERMES_IS_ANDROID)
+   add_definitions(-DHERMES_PLATFORM_UNICODE=HERMES_PLATFORM_UNICODE_JAVA)
+ endif()
+diff --git a/cmake/modules/Hermes.cmake b/cmake/modules/Hermes.cmake
+index 51abe346a..f28c12af6 100644
+--- a/cmake/modules/Hermes.cmake
++++ b/cmake/modules/Hermes.cmake
+@@ -429,3 +429,98 @@ if (GCC_COMPATIBLE)
+   check_cxx_compiler_flag("-Wrange-loop-analysis" RANGE_ANALYSIS_FLAG)
+   append_if(RANGE_ANALYSIS_FLAG "-Wno-range-loop-analysis" CMAKE_CXX_FLAGS)
+ endif (GCC_COMPATIBLE)
++
++# Based on
++# https://cristianadam.eu/20190501/bundling-together-static-libraries-with-cmake/
```

Comment:
this is basically taken over from 

https://cristianadam.eu/20190501/bundling-together-static-libraries-with-cmake/

with support for `$ARGN` so that we can pass in `jsi`